DiscordBot/bot.py

```diff
@@ -108,20 +113,41 @@ async def handle_channel_message(self, message):
             if message.author.id == self.user.id:
                 return
     
+            if not self.mod:
+                print("not mod")
+            print(self.mod)
             # Create an instance of ModeratorReport only once
-            moderator_report = ModeratorReport(self, message)
-
-            if message.content.startswith('!ban'):
-                # Call the handle_ban function from the ModeratorReport instance
-                await moderator_report.handle_ban(message)
-            elif message.content.startswith('!hide'):
-                # Call the handle_hide_profile function from the ModeratorReport instance
-                await moderator_report.handle_hide_profile(message)
-            elif message.content.startswith('!escalate'):
-                # Call the handle_escalate function from the ModeratorReport instance
-                await moderator_report.handle_escalate(message)
+            # Only respond to messages if they're part of a reporting flow
+            if not self.mod and not message.content.startswith(ModeratorReport.START_KEYWORD):
+                return
+
+            # If we don't currently have an active report for this user, add one
+            if not self.mod:
+                self.mod = ModeratorReport(client, message)
+            
+            moderator_report = self.mod
+            print(self.mod.state)
+            if message.content == ModeratorReport.START_KEYWORD:
+                # Let the report class handle this message; forward all the messages it returns to us
+                # TODO: get highest priority report
+                responses = await moderator_report.handle_report(message)
+                for r in responses:
+                    await message.channel.send(r)
             else:
-                await message.channel.send("Please enter a valid command: `!ban`, `!hide`, or `!escalate`.")
+                responses = await moderator_report.handle_report(message)
+                for r in responses:
+                    await message.channel.send(r)
+            # if message.content.startswith('!ban'):
+            #     # Call the handle_ban function from the ModeratorReport instance
+            #     await moderator_report.handle_ban(message)
+            # elif message.content.startswith('!hide'):
+            #     # Call the handle_hide_profile function from the ModeratorReport instance
+            #     await moderator_report.handle_hide_profile(message)
+            # elif message.content.startswith('!escalate'):
+            #     # Call the handle_escalate function from the ModeratorReport instance
+            #     await moderator_report.handle_escalate(message)
+            # else:
+            #     await message.channel.send("Please enter a valid command: `!ban`, `!hide`, or `!escalate`.")
```

Comment:
Should we remove the comments and prints before we merge?

Reply:
removed this and other comments and print statements I found